hub/core/chunk_engine/generator.py

```diff
@@ -0,0 +1,95 @@
+import numpy as np
+from typing import Generator, Tuple
+
+from hub.core.chunk_engine.exceptions import ChunkGeneratorError
+
+
+def chunk(
+    content_bytes: bytes, previous_num_bytes: int, chunk_size: int
+) -> Generator[Tuple[bytes, int], None, None]:
+    """
+    Generator function that chunks bytes.
+
+    Chunking is the process of taking the input `content_bytes` & breaking it up into smaller chunks,
+    the sizes of which are <= `chunk_size`.
+
+    Example 1:
+        content_bytes = b"101010101010"
+        chunk_size = 4
+        output_chunks = [b"1010", b"1010", b"1010"]
+
+        This is considered a "perfect fit" because all bytes fit perfectly into 3 chunks.
+
+
+    Example 2:
+        content_bytes = b"10101010111122"
```

Comment:
Unclear what base this is. Binary, ternary, hex, ASCII, I can't tell

Reply:
this is irrelevant, it's an arbitrary set of bytes

Reply:
to leave less room for interpretation, i will make it more random.